src/assets/utils/cardLinkCount.js

```diff
@@ -0,0 +1,13 @@
+export const cardLinkCount = (folders, links) => {
+  const linkCounts = Array(folders.length + 1).fill(0);
+  linkCounts[0] = links.length;
+  for (let link of links) {
+    for (let folderIndex = 0; folderIndex < folders.length; folderIndex++) {
+      if (link['folder_id'] === folders[folderIndex]) {
+        linkCounts[folderIndex + 1] += 1;
+      }
+    }
+  }
+
+  return linkCounts;
+};
```

Comment:
```suggestion
const getMatchedLinkCountList = (folders, links) => [
	links.length,
	...folders.map(
		(folder) => links.filter(({ folder_id: id }) => id === folder).length
	),
];
```

이런식으로 배열 내장메서드를 활용해보시는건 어떨까요?

현재 FE 트렌드가 명령형 프로그래밍보다는 선언형 프로그래밍을 추구하고 있고, 선언형 프로그래밍이 명령형보다 가독성이 훨씬 좋기 때문에 로직이 어떤 행동을 취하는지 해석하기보다 읽고 싶어하는 경향이 강합니다.

for 문이 절대 나쁘다고 하지는 못합니다.
성능상에서 미비한 차이기는 하지만 배열 내장메서드보다 for문이 성능측면에서 유리한것도 맞구요.

하지만 코드량이 많은 FE 특성상 가독성이 매우 중요한 요소로 작용합니다.

때문에 남이 읽기 편한 코드를 작성하는식으로 방향을 정해보시면 좋을듯합니다.

+ folders[folderIndex]는 객체인데 제대로 비교가 되나요?